### PR TITLE
Migrate student and groups step.

### PR DESCRIPTION
**Pull request by agorina**

### CODE REVIEW

File: migrate-reporting/src/test/java/org/opentestsystem/rdw/ingest/migrate/reporting/TestDataSourceConfiguration.java [hunk 1]
```diff
@@ -26,7 +26,7 @@
      *
      * @return DataSource
      */
-    @Bean(name = "batchDataSource")
+    @Bean(name = "datasource")
```
Comment (agorina) [inline migrate-reporting/src/test/java/org/opentestsystem/rdw/ingest/migrate/reporting/TestDataSourceConfiguration.java]: unrelated.

File: migrate-reporting/src/main/resources/application.staging.to.reporting.sql.yml [hunk 2]
```diff
@@ -431,21 +491,72 @@ sql:
               SET d.name = sd.name;
 
             delete: >-
-              DELETE FROM reporting.district
+              DELETE rd FROM reporting.district rd
                 WHERE id in (SELECT district_id from staging.staging_school WHERE deleted = 1)
-                    AND NOT EXISTS(SELECT id from reporting.school WHERE district_id = id);
+                    AND NOT EXISTS(SELECT id from reporting.school WHERE district_id = rd.id);
```
Comment (agorina) [inline migrate-reporting/src/main/resources/application.staging.to.reporting.sql.yml]: unrelated. some test found an issues here.

File: migrate-reporting/src/test/java/org/opentestsystem/rdw/ingest/migrate/reporting/TestDataSourceConfiguration.java (hunk 1 above)
Comment (chdietz) [inline migrate-reporting/src/test/java/org/opentestsystem/rdw/ingest/migrate/reporting/TestDataSourceConfiguration.java]: ok, makes sense.

Comment (chdietz) [inline migrate-reporting/src/test/java/org/opentestsystem/rdw/ingest/migrate/reporting/TestDataSourceConfiguration.java]: Reminds me on our real datasources such as for reporting connection with staging, reporting and the batch schema the connection could have not default.  
I was also thinking we could make the warehouse datasource read only to prevent accidental modifications.

Comment (agorina) [inline migrate-reporting/src/test/java/org/opentestsystem/rdw/ingest/migrate/reporting/TestDataSourceConfiguration.java]: Good point.
